<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="astro_proto">
  <material name="blue">
    <color rgba="0 0 0.7 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="silver">
    <color rgba="0.9 0.9 0.9 1"/>
  </material>
  <material name="brown">
    <color rgba="0.87 0.72 0.53 1"/>
  </material>

  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.760 0.470 0.780"/>A
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <link name="camera_support_link">
    <visual>
      <geometry>
        <cylinder length="0.115" radius="0.016"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.0575"/>
      <material name="brown"/>
    </visual>
  </link>
  <joint name="camera_support_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_support_link"/>
    <origin xyz="0 0 0.390"/>
  </joint>
  <link name="camera_link">
    <visual>
      <geometry>
        <box size="0.024 0.097 0.02"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.01"/>
      <material name="silver"/>
    </visual>
  </link>
  <joint name="camera_joint" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="camera_support_link"/>
    <child link="camera_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.115"/>
  </joint>
  <link name="left_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.0762"/>
      </geometry>
      <origin rpy="1.57075 0 0" xyz="0 -0.02 0"/>
      <material name="black"/>
    </visual>
  </link>
  <joint name="left_wheel_joint" type="continuous">
    <axis xyz="0 1 0"/>
    <parent link="base_link"/>
    <child link="left_wheel_link"/>
    <origin rpy="0 0 0" xyz="0 -0.134 -0.0385"/>
  </joint>
  <link name="right_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.0762"/>
      </geometry>
      <origin rpy="1.57075 0 0" xyz="0 0.02 0"/>
      <material name="black"/>
    </visual>
  </link>
  <joint name="right_wheel_joint" type="continuous">
    <axis xyz="0 1 0"/>
    <parent link="base_link"/>
    <child link="right_wheel_link"/>
    <origin rpy="0 0 0" xyz="0 0.134 -0.0385"/>
  </joint>
  <link name="support_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.03" radius="0.019"/>
      </geometry>
      <origin rpy="1.57075 0 0" xyz="0 0 0"/>
      <material name="black"/>
    </visual>
  </link>
  <joint name="support_wheel_joint" type="fixed">
    <!-- <axis xyz="0 1 1"/> -->
    <parent link="base_link"/>
    <child link="support_wheel_link"/>
    <origin rpy="0 0 0" xyz="-0.1 0 -0.0755"/>
  </joint>
</robot>